src/coreclr/System.Private.CoreLib/src/System/Array.CoreCLR.cs

```diff
@@ -381,8 +381,58 @@ private unsafe bool IsValueOfElementType(object value)
         // if this is an array of value classes and that value class has a default constructor
         // then this calls this default constructor on every element in the value class array.
         // otherwise this is a no-op.  Generally this method is called automatically by the compiler
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public extern void Initialize();
+        public unsafe void Initialize()
```

Comment:
@dotnet/ilc-contrib Should `Array.Initialize` be marked with `RequiresUnreferenceCode`, or is the value type default constructor in the special-cased by the trimmer and always preserved?

Reply:
I think it should:
```C#
var b = new ElementType[1];
b.Initialize();

struct ElementType
{
    public ElementType()
    {
        Console.WriteLine(".ctor called");
    }
}
```

`dotnet run` prints out ".ctor called".
But trimmed the app doesn't print out anything.

Related question - what should `default` do in this case, for example:
```C#
var c = new ElementType[1];
c[0] = default; // Should this call the default .ctor?
```

Running it seems like it will NOT call the .ctor - but what does it do?

Reply:
Same goes for AOT - published as Native AOT the app above also doesn't print out anything.

Reply:
> Related question - what should default do in this case, for example:

According to the specification, [default ignores the parameterless constructor and generates a zeroed instance](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/parameterless-struct-constructors#default-expression).

Reply:
Hmmm, adding `RequiresUnreferencedCode` on `Array.Initialize` has a ripple effect. It introduces warnings in situations where array type is passed into `DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)`. 

I guess it will need more careful thought.

Reply:
Opened #77426